juju/framework.py

```diff
@@ -218,6 +219,19 @@ def __get__(self, emitter, emitter_type):
 
     @classmethod
     def define_event(cls, event_kind, event_type):
+        """Define an event on this type at runtime.
+
+        cls -- a type to define an event on.
+        event_kind -- an attribute name that will be used to access the event. Must be a valid python identifier, not be a keyword or an existing attribute.
+        event_type -- a type of the event to define.
+        """
+        if not event_kind.isidentifier() or keyword.iskeyword(event_kind):
+            raise RuntimeError(f'unable to define an event with event_kind that is not a valid python identifier: {event_kind}')
+        elif keyword.iskeyword(event_kind):
+            raise RuntimeError(f'unable to define an event with event_kind that is a python keyword: {event_kind}')
+        elif event_kind in inspect.getmembers(cls):
```

Comment:
That doesn't look correct. It'll never trigger.

I suggest using getattr instead. But please add tests for all oif these cases before, so we can be sure this is all working as intended.